lib/modules/manager/terraform/hcl/schema.ts

```diff
@@ -0,0 +1,228 @@
+import { z } from 'zod';
+
+export type TfLiteral =
+  | string
+  | number
+  | boolean
+  | null
+  | { [k: string]: TfLiteral }
+  | TfLiteral[];
+
+export const tfLiteral: z.ZodType<TfLiteral> = z.lazy(() =>
+  z.union([
+    z.string(),
+    z.number(),
+    z.boolean(),
+    z.null(),
+    z.array(oneOrMany(tfLiteral)),
+    z.record(tfLiteral),
+  ]),
+);
+
+const NORMALISE_KEYS = new Set([
+  'metadata',
+  'spec',
+  'template',
+  'job_template',
+  'container',
+  'init_container',
+  'selector',
+]);
+
+function normalise(node: any): any {
```

Comment:
```suggestion
function normalise(node: unknown): unknown {
```
Do not use `any`.

If we are going this route we should do this for every block and not only the Kubernetes ones. 

> When a JSON object property is named after a nested block type, the value of this property represents one or more blocks of that type. The value of the property must be either a JSON object or a JSON array.

https://developer.hashicorp.com/terraform/language/syntax/json#nested-block-mapping